### PR TITLE
Add similarity score component

### PR DESCRIPTION
**Pull request by shivam-kapila**
This PR aims to add a similarityScore react component. The mockup looks as follows against various similarityScore values:

### < 0.3
image

### > 0.3 and < 0.7
image

### > 0.7
image

### Compact mode
image

P.S. No better name for the component came to my mind. Suggestions please.

### CODE REVIEW

File: listenbrainz/webserver/static/js/src/SimilarityScore.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      >
+        <div
+          className={`progress-bar ${className}`}
+          style={{
+            width: `${similarityScore * 100}%`,
+          }}
+        />
+      </div>
+      <p>
+        Your compatibility with {user.name} is {similarityScore * 10}/10!
```
Comment (MonkeyDo) [inline listenbrainz/webserver/static/js/src/SimilarityScore.tsx:43]:
```suggestion
        Your compatibility with {user.name} is {similarityScore * 10}/10
```

I like the enthusiasm, but I'd like to suggest a more toned-down version :)

File: listenbrainz/webserver/static/js/src/SimilarityScore.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        tabIndex={0}
+      >
+        <div
+          className={`progress-bar ${className}`}
+          style={{
+            width: `${similarityScore * 100}%`,
+          }}
+        />
+      </div>
+      <p>
```
Comment (MonkeyDo) [inline listenbrainz/webserver/static/js/src/SimilarityScore.tsx:42]: We might want a flag like a `showText` boolean in the props, and only return the text if it's set.
That way we can (as you suggested) show the similarity score in the "People similar to you" list.

Maybe the version without the "Your compatibility with…" phrase should have just `{similarityScore * 10}/10` next to or on the progress bar ?

**Review by MonkeyDo — COMMENTED** [review]
Haven't tried to render the component out anywhere but the code looks good.
Liking what I see on the screenshots!

File: listenbrainz/webserver/static/js/src/SimilarityScore.tsx (hunk 2 above)
Comment (shivam-kapila) [reply listenbrainz/webserver/static/js/src/SimilarityScore.tsx:42]: Your wish has been granted :p

**Review by paramsingh — COMMENTED** [review]
Code looks good to me, is there a way to possibly make the colors match up the LB theme more? Like purple instead of green or something like that?

**Review by paramsingh — APPROVED** [review]

**Review by MonkeyDo — APPROVED** [review]

**Pull request merged**